src/recommendation/src/test/scala/RankingTrainValidationSpec.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.recommendation.ALSModel
+import org.apache.spark.ml.util.MLReadable
+
+class RankingTrainValidationSplitSpec extends RankingTestBase with EstimatorFuzzing[RankingTrainValidationSplit] {
+
+  test("testALS") {
+
+    val tvRecommendationSplit = new RankingTrainValidationSplit()
+      .setEstimator(als)
+      .setEvaluator(evaluator)
+      .setEstimatorParamMaps(paramGrid)
+      .setTrainRatio(0.8)
+      .setUserCol(customerIndex.getOutputCol)
+      .setItemCol(itemIndex.getOutputCol)
+      .setRatingCol("rating")
+
+    val tvModel = tvRecommendationSplit.fit(transformedDf)
+
+    val model = tvModel.getBestModel.asInstanceOf[ALSModel]
+
+    val items = model.recommendForAllUsers(3)
```

Comment:
is this lazy or will this print/return always

Reply:
this is not lazy

Reply:
If it returns a dataframe then we should test that this dataframe should have the right structure